src/main/java/info/kgeorgiy/java/advanced/student/StudentDB.java

```diff
@@ -0,0 +1,169 @@
+package info.kgeorgiy.java.advanced.student;
+
+import java.util.*;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class StudentDB implements StudentQuery {
+
+    private static final String DEFAULT_EMPTY_STRING = "";
+
+    private enum Compare {
+        BY_NAME
+    }
+
+    private static final Map<Compare, Comparator<Student>> comparators = new HashMap<>() {{
+        put(Compare.BY_NAME, Comparator.comparing(Student::getLastName)
+                .thenComparing(Student::getFirstName)
+                .thenComparingInt(Student::getId));
+    }};
+
+    private String getFullName(Student s) {
+        return String.format("%s %s", s.getFirstName(), s.getLastName());
+    }
+
+    private Stream<Student> of(List<Student> students) {
+        return students.stream();
+    }
+
+    private Stream<Student> of(Collection<Student> students) {
```

Comment:
I am not sure that you need this. It is harder to understand than just common object.string()

Reply:
I am not sure as well, but it's sort of covering up all the stream things. It's like too straightforward and ugly to use word _stream_ to get `stream` and more elegant to just _get the stream`of` something_